test/jdk/java/awt/MenuItem/SetLabelTest.java

```diff
@@ -0,0 +1,183 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+import java.awt.Point;
+import java.awt.PopupMenu;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.io.File;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+
+/*
+ * @test
+ * @key headful
+ * @bug 4234266
+ * @summary To test setLabel functionality on Windows.
+ * @requires (os.family == "windows")
+ * @run main/othervm -Dsun.java2d.uiScale=1 SetLabelTest
+ */
+
+public class SetLabelTest implements ActionListener {
+    private static Robot robot;
+    private static Frame frame;
+    private static MenuBar mb;
+    private static Point frameLoc;
+    private static boolean passed = true;
+    private static final String[][] newLabels = {new String[]{"New Menu-1", "New MI-1"},
+                                                 new String[]{"New PM-1", "New PMI-1"}};
+
+    public static void main(String[] args) throws Exception {
+        SetLabelTest obj = new SetLabelTest();
+        obj.start();
+    }
+
+    public void start() throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(50);
+
+            EventQueue.invokeAndWait(this::createTestUI);
+            robot.delay(1000);
+            EventQueue.invokeAndWait(() -> frameLoc = frame.getLocationOnScreen());
+
+            // First Menu
+            robot.mouseMove(frameLoc.x + 35, frameLoc.y + 35);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.delay(300);
+            robot.mouseMove(frameLoc.x + 35, frameLoc.y + 90);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
```

Comment:
I don't think there's a need to click a menu item to make the label be changed.

In fact, the test fails for me on a High DPI display even with `-Dsun.java2d.uiScale=1`. The displayed menus are smaller, yet the title bar remains large and the robot misses the menu items. At the same time, robot is able to open both the first and the second menus if I run the test without `uiScale`, but it fails to click "Change" menu items (likely because the menu items aren't large enough for a High DPI display).